### PR TITLE
[FLINK-32628][tests][ci] Fix build_wheels_on_macos fails on AZP

### PR DESCRIPTION
**Pull request by snuyanzin**
The PR is aimed to fix build wheel on macos which starts failing
example of failure
https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=51555&view=logs&j=f73b5736-8355-5390-ec71-4dfdec0ce6c5&t=90f7230e-bf5a-531b-8566-ad48d3e03bbb&l=226
based on logs
```
2023-07-22T00:19:07.1527430Z       Collecting fastavro<2,>=0.23.6
2023-07-22T00:19:07.1543820Z         Downloading fastavro-1.8.0.tar.gz (792 kB)
```

it seems it fails to follow `dev-requirements.txt` where it's stated as 
```
fastavro>=1.1.0,<1.4.8
``` 
at the same time there `flink-python/pyproject.toml` containing `"apache-beam==2.43.0",` which depends on `fastavro` and seems started to violate requirements from `dev-requirements.txt`

The fix  just adds same dependency requirements to this file for `fastavro`

After that it was able to pass
example of pass on my azure pipeline
https://dev.azure.com/snuyanzin/flink/_build/results?buildId=2315&view=results

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: ( no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no )
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by snuyanzin** [comment]
current failure is not related to this PR and should be addressed within
https://issues.apache.org/jira/browse/FLINK-32632

**Comment by snuyanzin** [comment]
@dianfu  could you please have a  look
it's failing every nightly bulid

File: flink-python/pyproject.toml
Comment (snuyanzin) [inline flink-python/pyproject.toml:26]: version limitations are taken from 
https://github.com/apache/flink/blob/eab99b6de820352490cce580fee2e7f31ef16617/flink-python/dev/dev-requirements.txt#L28

**Comment by snuyanzin** [comment]
@XComp Could you please have a look
it's failing for every nigthly

**Comment by XComp** [comment]
The change looks reasonable. But maybe, let's wait for @dianfu's or @HuangXingBo's response. @PatrickRen do you have any other PyFlink developer who could jump in here to approve this change?

**Review by HuangXingBo — APPROVED** [review]
@snuyanzin Thanks a lot for the fix. LGTM. Could you help prepare a fix on release-1.16 too?

**Comment by snuyanzin** [comment]
Thanks for having a look
yes, sure, backports are created
once they pass ci will merge them

**Pull request closed without merging**
